crates/aptos/src/account/key_rotation.rs

```diff
@@ -0,0 +1,259 @@
+            new_private_key.sign_arbitrary_message(&rotation_msg.unwrap());
+
+        let txn_summary = self
+            .txn_options
+            .submit_transaction(aptos_stdlib::account_rotate_authentication_key_ed25519(
+                rotation_proof_signed_by_current_private_key
+                    .to_bytes()
+                    .to_vec(),
+                rotation_proof_signed_by_new_private_key.to_bytes().to_vec(),
+                // Existing public key
+                self.txn_options
+                    .private_key()?
+                    .public_key()
+                    .to_bytes()
+                    .to_vec(),
+                // New public key
+                new_private_key.public_key().to_bytes().to_vec(),
+            ))
+            .await
+            .map(TransactionSummary::from)?;
+
+        let string = serde_json::to_string_pretty(&txn_summary)
+            .map_err(|err| CliError::UnableToParse("trasaction summary", err.to_string()))?;
+
+        eprintln!("{}", string);
+
+        if let Some(txn_success) = txn_summary.success {
+            if !txn_success {
+                return Err(CliError::ApiError(
+                    "transaction was not executed successfully".to_string(),
+                ));
+            }
+        } else {
+            return Err(CliError::UnexpectedError(
+                "Mailformed transaction response".to_string(),
+            ));
+        }
+
+        // Asks user if they want to create a new Profile. Overriding profile is a bit of risky, we create a new one
+        // instead.
+
+        if let Err(cli_err) = prompt_yes_with_override(
+            "Do you want to create a profile with the new private key?",
+            self.prompt_options,
+        ) {
+            match cli_err {
+                CliError::AbortedError => {
+                    return Ok(());
+                }
+                _ => {
+                    return Err(cli_err);
+                }
+            }
+        }
+
+        eprintln!("Enter the name for the profile");
+        let profile_name = read_line("Profile name")?.trim().to_string();
+
+        if profile_name.is_empty() {
+            return Ok(());
+        }
+
+        let profile_config = ProfileConfig {
+            private_key: Some(new_private_key.clone()),
+            public_key: Some(new_private_key.public_key()),
+            ..self.txn_options.profile_options.profile()?
+        };
+
+        let mut config = CliConfig::load(ConfigSearchMode::CurrentDirAndParents)?;
+
+        if config.profiles.is_none() {
+            // This should not happen. The command requires a profile exist to rotate key
+            config.profiles = Some(BTreeMap::new());
+        }
+        config
+            .profiles
+            .as_mut()
+            .unwrap()
+            .insert(profile_name.clone(), profile_config);
+        config.save()?;
+
+        eprintln!("Profile {} is saved.", profile_name);
```

Comment:
This should be returned at the end with the transaction summary in a single object